### PR TITLE
NDEV-1925: Remove environmental, rewrite code

### PR DESCRIPTION
**Pull request by Deniskore**
1. Fully functional tracing feature (it did not work properly before)
2. The environmental crate has been removed.
3. Current changes allow EVM to be used in asynchronous/synchronous multithreaded code, current version in develop branch crashes, because crate environmental creates a global reference to a variable, which is unsafe.

### CODE REVIEW

**Review by dk-neon — CHANGES_REQUESTED** [review]

File: evm_loader/lib/src/commands/emulate.rs [hunk 1]
```diff
@@ -63,17 +68,18 @@ pub async fn execute(
         ..Transaction::default()
     };
 
-    let (exit_status, actions, steps_executed) = {
+    let (exit_status, actions, steps_executed, vm_trace, full_trace_data) = {
         let mut backend = ExecutorState::new(&storage);
         let mut evm = Machine::new(trx, tx_params.from, &mut backend)?;
 
+        let (vm_trace, full_trace_data) = evm.get_trace_data();
```
[the remark this answers is not in the document]
Comment (dk-neon) [reply evm_loader/lib/src/commands/emulate.rs]: Oh, now it's fixed.

**Review by anton-lisanin — CHANGES_REQUESTED** [review]
It's too intrusive in EVM code.
There is also tons of duplicated code.

**Comment by valiksinev** [comment]
neon-cli generates traces for all commands. So, tracing state-machine is tested by ci-tests of neon-evm, proxy (and uniswap).
The call of precompiled contracts raises error. The event EndStep is called without BeginStep (for derived trace instance).

Also, it is necessary to have only one instance of tracer for evm.

Instead:
let tracer = Rc::new(RefCell::new(Tracer::new()));
we should use the only one tracer instance

**Comment by Deniskore** [comment]
> It's too intrusive in EVM code. There is also tons of duplicated code.

It was duplicated before with macro, now everything is the same, only without the macro.

**Comment by Deniskore** [comment]
I was on vacation @andreisilviudragnea took this task for completion in

https://github.com/neonlabsorg/neon-evm/pull/149

**Pull request closed without merging**